### PR TITLE
added hw-21-1

### PR DESCRIPTION
**Pull request by kenu21**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/Factory.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
 import mate.academy.internetshop.dao.BucketDao;
 import mate.academy.internetshop.dao.ItemDao;
 import mate.academy.internetshop.dao.OrderDao;
 import mate.academy.internetshop.dao.UserDao;
 import mate.academy.internetshop.dao.impl.BucketDaoImpl;
-import mate.academy.internetshop.dao.impl.ItemDaoImpl;
+//import mate.academy.internetshop.dao.impl.ItemDaoImpl;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/Factory.java]: removed commented code from repository

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    private static final Logger logger = Logger.getLogger(ItemDaoJdbcImpl.class);
+    private static final String DB_NAME = "dbinternetshop";
+
+    public ItemDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public Item create(Item item) {
+        Statement stmt = null;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
        Statement statement = null;
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        String query = "Insert into " + DB_NAME + ".items (name, price) Values ("
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: try to use String.format() in all queries

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public Item get(Long id) {
+        Statement stmt = null;
+        String query = "Select * from " + DB_NAME + ".items where item_id = " + id + ";";
+
+        try {
+            stmt = connection.createStatement();
+            ResultSet rs = stmt.executeQuery(query);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
            ResultSet resultSet = stmt.executeQuery(query);
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    public Item create(Item item) {
+        Statement stmt = null;
+        String query = "Insert into " + DB_NAME + ".items (name, price) Values ("
+                + "'" + item.getName() + "'" + ", " + item.getPrice() + ");";
+        try {
+            stmt = connection.createStatement();
+            stmt.executeUpdate(query);
+            return item;
+        } catch (SQLException e) {
+            logger.warn("Can't create item", e);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
            logger.error("Can't create item", e);
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        } finally {
+            if (stmt != null) {
+                try {
+                    stmt.close();
+                } catch (SQLException e) {
+                    logger.warn("Can't close statement ", e);
+                }
+            }
+        }
+        return null;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]:
```suggestion
        return Collections.emptyList();
```

File: src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java [hunk 7]
```diff
@@ -51,7 +51,14 @@ public Bucket addItem(Long bucketId, Long itemId) {
     @Override
     public Bucket deleteItem(Long bucketId, Long itemId) {
         Bucket bucket = bucketDao.get(bucketId);
-        Item item = itemDao.get(itemId);
+        List<Item> list = bucket.getItems();
+        Item item = null;
+        for (Item it : list) {
+            if (itemId == it.getId()) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java]:
```suggestion
            if (itemId.equals(it.getId())) {
```

File: src/main/java/mate/academy/internetshop/Factory.java (hunk 1 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/Factory.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 2 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: OK

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 3 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 4 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 5 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java (hunk 6 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ItemDaoJdbcImpl.java]: OK.

File: src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java (hunk 7 above)
Comment (kenu21) [reply src/main/java/mate/academy/internetshop/service/impl/BucketServiceImpl.java]: OK

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**